components/CartSidebar.vue

```diff
@@ -0,0 +1,150 @@
+<template lang="pug">
+  div.cart-sidebar-wrapper(:class="openSidebarCart ? 'active' : ''")
+    div.overlay-cart-sidebar(@click="closeSidebarCart()")
+    aside.cart-sidebar
+        div.cart-sidebar-header
+            h4.title.is-3 Cart
+            img.lazyload(src="/cross.svg")(@click="closeSidebarCart()")
+        div.cart-sidebar-body
+          div(v-if="total > 0")
+            CartProductListItem(v-for="item in cart",
+                :key="item.name",
+                :item="item")
+          .empty.has-text-centered(v-else-if="!total")
+            h3 Your cart is empty.
+            nuxt-link(exact, :to="getHomeURL")
+              button.button(@click="closeSidebarCart()") Fill er up!
+        div.cart-sidebar-footer(v-if="total > 0")
+          p
+            b Total: {{ amount }} {{ currency }}
+          nuxt-link(exact, :to="getCartURL")
+            .button.is-primary(@click="closeSidebarCart()") Checkout
+</template>
+
+<script>
+import { createNamespacedHelpers } from "vuex"
+import Switch from "@/components/Switch"
+import CartProductListItem from "@/components/CartProductListItem"
+
+const { mapGetters: cartGetters } = createNamespacedHelpers("cart")
+
+export default {
+  name: "CartSidebar",
+  components: {
+    AppSwitch: Switch,
+    CartProductListItem,
+  },
+  computed: {
+    ...cartGetters(["openSidebarCart", "cart", "total", "amount"]),
+    getCartURL() {
+      const storeID = this.$route.params.id
+      return storeID
+        ? { name: "store-id-cart", params: { id: storeID } }
+        : { name: "cart" }
+    },
+    currency() {
+      return this.$store.state.store.default_currency
+    },
+    getHomeURL() {
+      const storeID = this.$route.params.id
+      return storeID
+        ? { name: "store-id", params: { id: storeID } }
+        : { name: "index" }
+    },
+  },
+  methods: {
+    closeSidebarCart() {
+      this.$store.dispatch("cart/openSidebarCart", false)
+    },
+  },
+}
+</script>
+
+<style lang="scss" scoped>
```

Comment:
I don't understand this css much but all I can tell is: it looks beautiful :smile: